homeassistant/components/whirlpool/__init__.py

```diff
@@ -30,7 +31,7 @@ async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
     auth = Auth(backend_selector, entry.data[CONF_USERNAME], entry.data[CONF_PASSWORD])
     try:
         await auth.do_auth(store=False)
-    except aiohttp.ClientError as ex:
```

Comment:
Why is `ClientError` changing to `ClientConnectionError`? I don't think this should change... But if you have a reason to change this existing code, make it a separate PR that explains this bug fix rather than making it a part of adding new sensors?

Reply:
I have another integration that's this way, and it looks like several other integrations use these two exceptions.  Now that I'm looking around I see some integrations use ClientError to throw a connection error.  Not clear to me which is "right" or better....
I'll switch it back...

Reply:
OK, to elaborate...

`ClientError` is the base class and `ClientConnectionError` is a more specific exception. `ClientConnectionError` could be an issue at the socket level.`ClientError` also could include other errors like a `ClientResponseError` that the device returns -- e.g. "bad request", or some other failure. `TimeoutError` is due to not getting a response within a certain amount of time.

Conformity across integrations is great and all, but changing things to match without having a specific technical reason is not the goal.

Thanks